gluten-core/src/main/scala/io/glutenproject/expression/StringExpressionTransformer.scala

```diff
@@ -0,0 +1,110 @@
+
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.expression.ConverterUtils.FunctionConfig
+import io.glutenproject.substrait.expression._
+
+import org.apache.spark.sql.catalyst.expressions._
+
+import com.google.common.collect.Lists
+
+case class String2TrimExpressionTransformer(
+    substraitExprName: String,
+    trimStr: Option[ExpressionTransformer],
+    srcStr: ExpressionTransformer,
+    original: Expression)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val trimStrNode = trimStr.map(_.doTransform(args))
+    val srcStrNode = srcStr.doTransform(args)
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionName =
+      ConverterUtils.makeFuncName(
+        substraitExprName,
+        original.children.map(_.dataType),
+        FunctionConfig.REQ)
+    val functionId = ExpressionBuilder.newScalarFunction(functionMap, functionName)
+    val expressNodes = Lists.newArrayList[ExpressionNode]()
+    trimStrNode.foreach(expressNodes.add)
+    expressNodes.add(srcStrNode)
+    val typeNode = ConverterUtils.getTypeNode(original.dataType, original.nullable)
+    ExpressionBuilder.makeScalarFunction(functionId, expressNodes, typeNode)
+  }
+}
+
+case class StringTranslateTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    matchingExpr: ExpressionTransformer,
+    replaceExpr: ExpressionTransformer,
+    original: StringTranslate)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    // In CH, translateUTF8 requires matchingExpr and replaceExpr argument have the same length
+    val matchingNode = matchingExpr.doTransform(args)
+    val replaceNode = replaceExpr.doTransform(args)
+    if (
+      !matchingNode.isInstanceOf[StringLiteralNode] ||
+      !replaceNode.isInstanceOf[StringLiteralNode]
+    ) {
+      throw new UnsupportedOperationException(s"$original not supported yet.")
+    }
+
+    val matchingLiteral = matchingNode.asInstanceOf[StringLiteralNode].getValue
+    val replaceLiteral = replaceNode.asInstanceOf[StringLiteralNode].getValue
+    if (matchingLiteral.length() != replaceLiteral.length()) {
+      throw new UnsupportedOperationException(s"$original not supported yet.")
+    }
+
+    NaryExpressionTransformer(substraitExprName, Seq(srcExpr, matchingExpr, replaceExpr), original)
+      .doTransform(args)
+  }
+}
+
+case class RegExpReplaceTransformer(
+    substraitExprName: String,
+    subject: ExpressionTransformer,
+    regexp: ExpressionTransformer,
+    rep: ExpressionTransformer,
+    pos: ExpressionTransformer,
+    original: RegExpReplace)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.isVeloxBackend) {
+      NaryExpressionTransformer(substraitExprName, Seq(subject, regexp, rep, pos), original)
```

Comment:
Looks a historical issue. Shouldn't we add `return` for this statement?

BTW, I think `nary` indicates there is no child. In contrast, unary/binary respectively for one/two child/children may be derived from this word. The name `NaryExpressionTransformer` may not correctly reflect its usage. Right?
Maybe, `GenericExpressionTransformer` is better to indicate there is no special handling to transform. Any other proposal? cc @rui-mo

Reply:
>  Shouldn't we add return for this statement?

Great catch!

And `Nary` what I'm trying to mean is `N-ary`, it seems a bit ambiguous, I agree with using `GenericExpressionTransformer `.